test/db_tests/db_test.js

```diff
@@ -0,0 +1,218 @@
+                expect(results).to.equal(false);
+                done();
+                endDate = '2000-01-02'
+            });
+        });
+    });
+
+    describe('getUser + two dates + out of range', function(){
+        it('Should return empty set of data', function(done){
+            controllers.getUsers(startDate, endDate).done(function(results){
+                expect(results).to.have.length(0);
+                done();
+                startDate = '2018-02-02';
+                endDate = '2018-02-04';
+            });
+        });
+    });
+
+    describe('getUser + two dates + in range', function(){
+        it('Should return subset of data', function(done){
+            controllers.getUsers(startDate, endDate).done(function(results){
+                expect(results).to.have.length.above(0);
+                done();
+            });
+        });
+    });
+
+/*
+     -------------Note-------------
+     Please Write any new tests above the modifyUser Tests
+     as the last modifyUser test calls the process.exit() command
+     to close DB connection.
+
+     Thanks!
+ */
+
+/*
+    -------------modifyUser Tests-------------
+*/
+
+    describe('modifyUser invalid badge id', function(){
+        it('Should not update and return status false', function(done){
+            controllers.modifyUser('fake badge id', {'password': 'fake'}).done(function(results){
+                expect(results.result).to.equal(false);
+                done();
+            });
+        });
+    });
+
+    describe('modifyUser + valid badge id + valid attribute field + null value', function(){
+        it('Should not update table.  Desired update attribute cannot be null.  Return status false', function(done){
+            controllers.modifyUser('1234567', {'first': null}).done(function(results){
+                expect(results.result).to.equal(false);
+                done();
+            });
+        });
+    });
+
+    describe('modifyUser + valid badge id + valid attribute field + null value', function(){
+        it('Should update table.  Desired attribute can be null.  Return status true', function(done){
+            controllers.modifyUser('1234567', {'password': null}).done(function(results){
+                expect(results.result).to.equal(true);
+                done();
+            });
+        });
+    });
+
+    describe('modifyUser + valid badge id + valid attribute field + value', function(){
+        it('Should update table attribute, return status true', function(done){
+            controllers.modifyUser('1234567', {'password': 'fart'}).done(function(results){
+                expect(results.result).to.equal(true);
+                done();
+            });
+        });
+    });
+
+    describe('modifyUser + valid badge id + invalid attribute field', function(){
+        it('Should not update table attribute.  Return status true', function(done){
+            controllers.modifyUser('1234567', {'passowrd': 'lalala'}).done(function(results){
```

Comment:
Why does a wrong attribute name return true? Shouldn't it return false? I just feel like if someone accidentally misspelled an attribute name and its returned `true` then they may have a wrong impression that the function actually worked and modified that value.

Reply:
I'm not 100% on this.  Yes your logic is correct, as it was also my assumption when first running this test that is should fail.  I know from watching the data that is does not update, but for some reason when passing in the attribute passowrd, or even just an empty JSON object, sequelize doesn't catch this as an error and instead returns a result of [1] which is the same result it returns for a successful update.

--I get the same results when manually entering in the wrong attributes as well.

As for why this happens?  No entirely sure.  I've been looking into but haven't found much of a solid answer.